reconcile/saas_auto_promotions_manager/subscriber.py

```diff
@@ -118,6 +122,17 @@ def _passed_accumulated_soak_days(self) -> bool:
                 delta += now - deployed_at
         return delta >= timedelta(days=self.soak_days)
 
+    def _is_valid_deployment_window(self) -> bool:
+        # Ideally we would catch that at schema validation time
+        if not croniter.is_valid(self.schedule):
+            logging.error(
+                "Subscriber at %s has an invalid schedule declaration %s. We will block any promotion for that target until this is fixed.",
+                self.target_file_path,
+                self.schedule,
+            )
+            return False
+        return croniter.match(self.schedule, datetime.now(UTC))
```

Comment:
@fishi0x01 How do we ensure there is an integration run at the matching time? For example `1 * * * *` to run on min 1 every hour, but `datetime.now(UTC)` may miss that if the integration finish at min 0 then sleep for 10 mins, miss that time window.

Reply:
Auto-promotion windows are more tailored towards bigger time windows, e.g., only promote during working hours.

Reply:
then we need to doc or limit that so no unexpected schedule miss.